source/common/buffer/buffer_util.cc

```diff
@@ -0,0 +1,43 @@
+#include "source/common/buffer/buffer_util.h"
+
+#include <charconv>
+
+#include "source/common/common/macros.h"
+
+namespace Envoy {
+namespace Buffer {
+
+void Util::serializeDouble(double number, Buffer::Instance& buffer) {
```

Comment:
Consider using this in the prometheus histogram output also. Current code: 

```
output.append(fmt::format("{0}_bucket{{{1}le=\"{2:.32g}\"}} {3}\n", prefixed_tag_extracted_name,
                              hist_tags, bucket, value));
```

Reply:
I would predict switching to this API would make that particular code slower. I think this new mechanism works well in the context of json streaming because it's just doing one object. But I think if I have to atomize a fmt::format call like that I will lose more performance.

Also this API appends to a buffer, and this is in the context of formatting a much larger string, which will later be appended to a buffer. So to even use this cleanly will require a significant refactor.

I think I'd prefer not to go this route.